web/src/buddy_data.ts

```diff
@@ -363,7 +363,7 @@ function maybe_shrink_list(
 
     // We want to always show PM recipients even if they're inactive.
     const pm_ids_set = narrow_state.pm_ids_set();
-    const stream_id = narrow_state.stream_id();
+    const stream_id = narrow_state.stream_id(narrow_state.filter(), true);
```

Comment:
We could also call this with `undefined` for the `Filter` parameter, but that seemed like it might be confusing when reading the code.